validation-test/stdlib/Arrays.swift.gyb

```diff
@@ -138,18 +138,229 @@ ArrayTestSuite.test("${Kind}/popLast") {
+  }
+}
+% end
+
+ArrayTestSuite.test("ArraySlice/append") 
+  .xfail(.custom({true}, reason: "<rdar://problem/33997561>")).code {
+  do {
+    let a: [OpaqueValue<Int>] = [ 42, 2525, 3535, 42 ].map(OpaqueValue.init)
+    var s = a[1..<2]
+    let v = OpaqueValue(1515)
+    s.append(v)
+    expectEqual(1, s.startIndex)
+    expectEqual(3, s.endIndex)
+  }
+}
+
+// Check how insert(_:at:) affects indices.
+% for Kind in ['Array', 'ContiguousArray']:
+ArrayTestSuite.test("${Kind}/insert") {
+  do {
+    var a: ${Kind}<OpaqueValue<Int>> = ${Kind}([ 2 ].map(OpaqueValue.init))
+    let v = OpaqueValue(1)
+    a.insert(v, at: 0)
+    expectEqual(0, a.startIndex)
+    expectEqual(2, a.endIndex)
+  }
+  do {
+    var a: ${Kind}<OpaqueValue<Int>> = ${Kind}([ 2 ].map(OpaqueValue.init))
+    let v = OpaqueValue(1)
+    a.insert(v, at: 1)
+    expectEqual(0, a.startIndex)
+    expectEqual(2, a.endIndex)
+  }
+}
+% end
+
+ArrayTestSuite.test("ArraySlice/insert") {
+  do {
+    let a: [OpaqueValue<Int>] = [ 42, 2525, 3535, 42 ].map(OpaqueValue.init)
+    var s = a[1..<2]
+    let v = OpaqueValue(1515)
+    s.insert(v, at: 1)
+    expectEqual(1, s.startIndex)
+    expectEqual(3, s.endIndex)
+  }
+}
+
+// Check how partition(by:) affects indices.
+% for Kind in ['Array', 'ContiguousArray']:
+ArrayTestSuite.test("${Kind}/partition") {
+  do {
+    var a: ${Kind}<OpaqueValue<Int>> = ${Kind}([ 10, 2, -33, 44, -5 ].map(OpaqueValue.init))
+    a.partition(by: { $0.value > 0 })
+    expectEqual(0, a.startIndex)
+    expectEqual(5, a.endIndex)
+  }
+  do {
+    var a: ${Kind}<OpaqueValue<Int>> = ${Kind}([ 10, 2, -33, 44, -5 ].map(OpaqueValue.init))
+    a.partition(by: { $0.value > 100 })
+    expectEqual(0, a.startIndex)
+    expectEqual(5, a.endIndex)
+  }
+  do {
+    var a: ${Kind}<OpaqueValue<Int>> = ${Kind}([ 10, 2, -33, 44, -5 ].map(OpaqueValue.init))
+    a.partition(by: { $0.value > 100 })
+    expectEqual(0, a.startIndex)
+    expectEqual(5, a.endIndex)
+  }
+}
+% end
+
+ArrayTestSuite.test("ArraySlice/partition") {
+  do {
+    let a: [OpaqueValue<Int>] = [ 99, 1010, 99 ].map(OpaqueValue.init)
+    var s = a[1..<2]
+    s.removeFirst()
+    expectEqual(2, s.startIndex)
+    expectEqual(2, s.endIndex)
+  }
+  do {
+    let a: [OpaqueValue<Int>] = [ 99, 1010, 2020, 99 ].map(OpaqueValue.init)
+    var s = a[1..<3]
+    s.removeFirst()
+    expectEqual(2, s.startIndex)
+    expectEqual(3, s.endIndex)
+  }
+}
+
+// Check how removeLast() affects indices.
+% for Kind in ['Array', 'ContiguousArray']:
+ArrayTestSuite.test("${Kind}/removeLast") {
+  do {
+    var a: ${Kind}<OpaqueValue<Int>> = ${Kind}([ 1 ].map(OpaqueValue.init))
+    a.removeLast()
+    expectEqual(0, a.startIndex)
+    expectEqual(0, a.endIndex)
+  }
+  do {
+    var a: ${Kind}<OpaqueValue<Int>> = ${Kind}([ 1, 2 ].map(OpaqueValue.init))
+    a.removeLast()
+    expectEqual(0, a.startIndex)
+    expectEqual(1, a.endIndex)
+  }
+}
+% end
+
+ArrayTestSuite.test("ArraySlice/removeLast") {
```

Comment:
I'd also test the case of an empty slice. Both before and after calling  `removeLast`.